dora/core/server/worker/src/main/java/alluxio/worker/modules/WorkerIdentityProvider.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.modules;
+
+import alluxio.conf.AlluxioConfiguration;
+import alluxio.conf.PropertyKey;
+import alluxio.wire.WorkerIdentity;
+
+import com.google.common.collect.Sets;
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.PosixFilePermission;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+/**
+ * Provider that resolves worker identities from configuration, persisted storage, or
+ * automatically generates a new one.
+ */
+public class WorkerIdentityProvider implements Provider<WorkerIdentity> {
+  private static final Logger LOG = LoggerFactory.getLogger(WorkerIdentityProvider.class);
+  private static final String WORKER_IDENTITY_FILE = "worker_identity";
+  private final AlluxioConfiguration mConf;
+
+  /**
+   * @param conf configuration
+   */
+  @Inject
+  public WorkerIdentityProvider(AlluxioConfiguration conf) {
+    mConf = conf;
+  }
+
+  /**
+   * Resolves a worker's identity from the following sources, in the order of preference:
+   * <ol>
+   *   <li>
+   *     Alluxio configuration. This includes JVM system properties, alluxio-site.properties, etc.
+   *   </li>
+   *   <li>
+   *     Persisted storage. A identity file is persisted in the worker process's working directory.
+   *   </li>
+   *   <li>
+   *     Automatically generated as a UUID.
+   *   </li>
+   * </ol>
+   *
+   * @return worker identity
+   */
+  @Override
+  public WorkerIdentity get() {
+    // Look at configurations first
+    if (mConf.isSetByUser(PropertyKey.WORKER_IDENTITY_UUID)) {
+      String uuidStr = mConf.getString(PropertyKey.WORKER_IDENTITY_UUID);
+      return WorkerIdentity.ParserV1.INSTANCE.fromUUID(uuidStr);
+    }
+
+    // Try loading from the identity file
+    String workDir = mConf.getString(PropertyKey.WORK_DIR);
```

Comment:
WORK_DIR is not persisted in k8s. The only persisted place is the worker cache. Does it make sense to put the identity file into page cache?

Reply:
no wait, it cant be in page cache, upon restart, unrecognized formated file will be deleted. we can put in preserved dir such as logs/ journal/ -> @ssz1997  actually which dirs under alluxio are considered preserved dirs? does conf/ gets preserved?

Reply:
Journal does get preserved but it's only for master.
log/ can be preserved. It's a configuration if user wants to preserve log. By default it is preserved but I can't guarantee all users can do this.
conf/ doesn't get preserved.

Reply:
yeah then we need a safe place to store this workerid file for this to work in the first place  @beinan  @apc999

Reply:
If in k8s, why do you need to persist the worker id to a local file? Kubernetes should be responsible for creating the worker identity UUID at the time of creating the worker instance, and associate that identity with it until it is destroyed. The property key `alluxio.worker.identity.uuid` is exactly for this purpose.

Reply:
K8s doesn't have a place to store the worker identity - it's just an ETCD cluster. The operator is completely stateless and has no persisted storage. So k8s cannot tell if a new worker is actually new or is a restarted old worker just by itself, because in both cases the container ip is new. We have to count on worker informing the etcd its identity.

Reply:
> If in k8s, why do you need to persist the worker id to a local file? Kubernetes should be responsible for creating the worker identity UUID at the time of creating the worker instance, and associate that identity with it until it is destroyed. The property key `alluxio.worker.identity.uuid` is exactly for this purpose.

oh i got what u r saying, so it could plugin the `alluxio.worker.identity.uuid` at time of helm install u mean to say?

Reply:
@dbw9580  so i syned up offline with Shawn, it doenst seem possible to generate a UUID at deployment time to plug into alluxio site properties, maybe we need to rely on the workerid file, can three of us sync offline on this to unblock Shawn on his k8s operator side of change?

Reply:
@dbw9580 Yeah currently there's no easy way to inject UUID into each worker, and k8s is unable to tell if a newly started worker is a new worker or a registered worker, unless the worker tells it so.
If storing local file is not feasible in the worker cache, we may need a different directory to store it. I would suggest make this a property key.

Reply:
making the path of the identity file configurable sounds like a good idea.